les/txrelay.go

```diff
@@ -121,7 +121,10 @@ func (self *LesTxRelay) send(txs types.Transactions, count int) {
 				return peer.GetRequestCost(SendTxMsg, len(ll))
 			},
 			canSend: func(dp distPeer) bool {
-				return dp.(*peer) == pp
+				if !dp.(*peer).isOnlyAnnounce {
```

Comment:
Why not use a single expression? Lazy evaluation should compile it to the exact same code.
return !dp.(*peer).isOnlyAnnounce && dp.(*peer) == pp